File: java/CodeForces/src/Pogoda.java
// codeforces 234 C done but CodeForces hamiat, "Ошибка времени исполнения на тесте 1"
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;


public class Pogoda {
	int n;
	int []t;
	int []nn;
	int []np;
	
	public Pogoda() throws IOException {
		BufferedReader bin = new BufferedReader(new InputStreamReader(System.in));
		n = Integer.parseInt(bin.readLine());
		t = new int[n];
		String s = bin.readLine();
		String []ss = s.split(" ");
		for (int i=0; i<n; i++) {
			t[i] = Integer.parseInt(ss[i]);
		}		
		nn = new int[n];
		np = new int[n];
		bin.close();
	}
	
	void countNeg() {
		int cnt = 0;
		for (int i=0; i<n; i++) {
			if (t[i] >= 0) cnt++;
			nn[i] = cnt;			
		}
	}
	
	void countPos() {
		int cnt = 0;
		for (int i=n-1; i>=0; i--) {
			if (t[i] <= 0) cnt++;
			nn[i] = cnt;			
		}
	}
	
	int findOpt() {
		int oo = Integer.MAX_VALUE;
		for (int i=0; i<n-1; i++) {
			int o = nn[i] + np[i+1];
			if (o < oo) {
				oo = o;
			}					
		}
		return oo;
	}

	/**
	 * @param args
	 */
	public static void main(String[] args) throws IOException {
		Pogoda p = new Pogoda();
		p.countNeg();
		p.countPos();
		int oo = p.findOpt();
		System.out.println(oo);
	}
}

File: scala/misscal/src/Pogoda.scala
// CodeForces 234 C

object Pogoda extends App {
  var s = Console.readLine()
  var ss = (s.split(" ")) map (_.toInt)
  val n = ss(0)
  s = Console.readLine()
  ss = (s.split(" ")) map (_.toInt)
  val t = ss map (_.toInt)
  var nn: Array[Int] = Array.fill(n)(0)
  var np: Array[Int] = Array.fill(n)(0)

  def countNeg(i: Int, cntNeg: Int): Unit = {
    if (i == n) ()
    else {
      val cn = if (t(i) >= 0) cntNeg + 1 else cntNeg
      nn(i) = cn
      countNeg(i + 1, cn)
    }
  }

  def countPos(i: Int, cntPos: Int): Unit =
    if (i < 0) ()
    else {
      val cpU = if (t(i) <= 0) cntPos + 1 else cntPos
      np(i) = cpU
      countPos(i - 1, cpU)
    }

  def findOpt(i: Int, lastOpt: Int): Int =
    if (i == n-1) lastOpt
    else {
      val o = nn(i) + np(i+1)
      findOpt(i + 1, if (o < lastOpt) o else lastOpt)
    }

  countNeg(0, 0)
  countPos(n - 1, 0)
  val oo = findOpt(0, 1000000000)
  println(oo)
}